pkg/controller/build/reconciler.go

```diff
@@ -791,6 +911,150 @@ func (b *buildReconciler) deleteMachineOSBuild(ctx context.Context, mosb *mcfgv1
 	return fmt.Errorf("could not delete MachineOSBuild %s for MachineOSConfig %s: %w", mosb.Name, moscName, err)
 }
 
+func (b *buildReconciler) getAuthFilePath(mosb *mcfgv1.MachineOSBuild, moscName string) (string, error) {
+	pushSecret := mosb.GetAnnotations()[constants.RenderedImagePushSecretAnnotationKey]
+	secret, err := b.kubeclient.CoreV1().Secrets(ctrlcommon.MCONamespace).Get(context.TODO(), pushSecret, metav1.GetOptions{})
```

Comment:
suggestion (non-blocking): I think we have the secrets lister present here. If so, use that to retrieve the secret instead of the API client.

Reply:
I checked and the secret lister is not present here